[executorch] Add support for method variant functions in ExecuTorch codegen

JacobSzwejbka opened the pull request:
Summary: Title.

Test Plan: The added unittest

Differential Revision: D54021036

larryliu0820 commented on torchgen/gen_executorch.py:
```diff
@@ -123,10 +123,15 @@ class ComputeFunction:
         if not self.selector.is_root_operator(f"{f.namespace}::{f.func.name}"):
             return None
         if Variant.function not in f.variants:
-            return None
+            if Variant.method in f.variants:
+                is_method_variant = True
+            else:
+                return None
+
```
Can we error out in this case? It took us a while to understand what's going on. We also want to error out if both method and function are in variants.

larryliu0820 commented on tools/test/test_executorch_gen.py:
```diff
@@ -332,6 +332,18 @@ def setUp(self) -> None:
+            custom_3_backend_index,
+        ) = NativeFunction.from_yaml(
+            {
+                "func": "custom_3::op_3(Tensor(a!) self) -> Tensor(a!)",
+                "dispatch": {"CPU": "kernel_3"},
```
Can this function take some arguments so that we have more coverage?

larryliu0820 reviewed (approved):
Looks Good

JacobSzwejbka committed:
[executorch] Add copy_ to aten op lib.

Summary: Title.

Test Plan: unittest

Reviewed By: larryliu0820

Differential Revision: D54021036
